katello/katello/hostname-change.rb

```diff
@@ -209,6 +231,24 @@ def setup_opt_parser
     def run
       raise 'Must run as root' unless Process.uid == 0
 
+      scenario_answers = load_scenario_answers(@options[:scenario])
+
+      if using_custom_certs?(scenario_answers)
+        unless all_custom_cert_options_present?(scenario_answers)
+          fail_with_message("You are currently using custom certificates but not all custom " \
+                            "certificate arguments are present. To change the hostname, " \
+                            "you need to provide certificates with the new hostname. Please " \
+                            "review the custom certificate arguments required with --help.")
+        end
+      end
+
+      if using_custom_certs?(scenario_answers) && !@foreman_proxy_content
```

Comment:
Can you elaborate why the certs only need to be checked in non-proxy context?

Reply:
Yes, the foreman proxy hostname change requires a new certs-tarball whether the user is using custom certs or not. This tarball is passed into k-c-h as a single argument and custom certs are included in this tarball. 

Unfortunately there isn't a good way to check the certs in the tarball right now, we should make katello-certs-check work for proxies and check these sorts of things in the future.